source/javascripts/services/request-service.ts

```diff
@@ -0,0 +1,158 @@
+
+	constructor(logger: Logger) {
+		this.configure({ logger });
+	}
+
+	public configure({ mode, appSlug, logger }: RequestServiceConfigureOptions = {}): void {
+		this.mode = mode || (this.modeFromEnvVars() as string);
+		this.appSlug = appSlug || (getAppSlug() as string);
+		this.logger = logger;
+	}
+
+	public async getAppConfigYML(
+		abortedPromise: Promise<undefined>
+	): Promise<string | Error | { bitrise_yml: string; error_message: Error }> {
+		const controller = new AbortController();
+		if (abortedPromise) {
+			abortedPromise.then(() => {
+				controller.abort();
+			});
+		}
+
+		let requestURL = "";
+
+		switch (this.mode) {
+			case RequestServiceMode.Website:
+				requestURL = StringService.stringReplacedWithParameters(window["routes"].website.yml_get, {
+					app_slug: this.appSlug // eslint-disable-line @typescript-eslint/camelcase
+				});
+
+				break;
+			case RequestServiceMode.Cli:
+				requestURL = window["routes"].local_server.yml_get;
+
+				break;
+		}
+
+		let response: Response;
+		try {
+			response = await fetch(requestURL, {
+				signal: controller.signal
+			});
+		} catch (error) {
+			this.logErrorWithLevel(error, StatusType.warn);
+			throw this.requestAbortedError(window["strings"].request_service.load_app_config.error_prefix);
+		}
+
+		if (!response.ok) {
+			let responseBody: { bitrise_yml?: string; error: string };
+			try {
+				responseBody = await response.json();
+			} catch (error) {
+				this.logErrorWithLevel(error, StatusType.error);
+				throw Error(window["strings"].request_service.load_app_config.default_error);
+			}
+
+			if (responseBody.bitrise_yml) {
+				const error = this.prefixedError(
+					this.errorFromResponseBody(
+						responseBody,
+						window["strings"].request_service.load_app_config.invalid_bitrise_yml_error
+					).message,
+					window["strings"].request_service.load_app_config.error_prefix
+				);
+				this.logErrorWithLevel(error, StatusType.warn);
+
+				throw {
+					bitriseYml: responseBody.bitrise_yml,
+					error: error
+				};
+			}
+
+			const error = this.errorFromResponseBody(
+				responseBody,
+				window["strings"].request_service.load_app_config.default_error
+			);
+			this.logErrorWithLevel(error, response.status < 500 ? StatusType.warn : StatusType.error);
+			throw error;
+		}
+
+		try {
+			return await response.text();
+		} catch (error) {
+			this.logErrorWithLevel(error, StatusType.error);
+			throw new Error(window["strings"].request_service.load_app_config.default_error);
+		}
+	}
+
+	private modeFromEnvVars(): RequestServiceMode {
+		if (process.env["MODE"] === "WEBSITE") {
+			return RequestServiceMode.Website;
+		}
+
+		return RequestServiceMode.Cli;
+	}
+
+	private prefixedError(message: string, messagePrefix: string): Error {
+		return new Error(messagePrefix + message);
+	}
+
+	private requestAbortedError(messagePrefix: string): Error {
+		return this.prefixedError(window["strings"].request_service.request.aborted, messagePrefix);
```

Comment:
I'd put the error string here instead of fetching it from a file, I don't think we want to use it outside of request service.

Reply:
Same as with routes, I think we should do it in a separate iteration.